src/stcal/ramp_fitting/ols_fit.py

```diff
@@ -363,174 +149,422 @@ def ols_ramp_fit_sliced(
+def get_integ_slice(integ_info, integ_slice, row_start, nrows):
+    """
+    Populates the integration output information from each slice.
 
-    opt_pedestal : ndarray
-        The zero point for each pixel in each segment of each integration, 4-D float32
+    integ_info: tuple
+        The output integration information to populate from the slice.
 
-    opt_weights : ndarray
-        The weight of each pixel to use in combining the segments, 4-D float32
+    integ_slice: tuple
+        The output slice used to populate the output arrays.
 
-    opt_crmag : ndarray
-        The magnitude of each CR in each integration, 4-D float32
+    row_start: int
+        The start row the current slice at which starts.
+
+    nrows: int
+        The number of rows int the current slice.
+    """
+    data, dq, var_poisson, var_rnoise, int_times, err = integ_info
+    idata, idq, ivar_poisson, ivar_rnoise, iint_times, ierr = integ_slice
+
+    srow, erow = row_start, row_start + nrows
+
+    data[:, srow:erow, :] = idata
+    dq[:, srow:erow, :] = idq
+    var_poisson[:, srow:erow, :] = ivar_poisson
+    var_rnoise[:, srow:erow, :] = ivar_rnoise
+    err[:, srow:erow, :] = ierr
+
+
+def get_opt_slice(opt_info, opt_slice, row_start, nrows):
+    """
+    Populates the optional output information from each slice.
+
+    opt_info: tuple
+        The output optional information to populate from the slice.
+
+    opt_slice: tuple
+        The output slice used to populate the output arrays.
+
+    row_start: int
+        The start row the current slice at which starts.
+
+    nrows: int
+        The number of rows int the current slice.
+    """
+    (slope, sigslope, var_poisson, var_rnoise,
+        yint, sigyint, pedestal, weights, crmag) = opt_info
+    (oslope, osigslope, ovar_poisson, ovar_rnoise,
+        oyint, osigyint, opedestal, oweights, ocrmag) = opt_slice
+
+    srow, erow = row_start, row_start + nrows
+
+    # The optional results product is of variable size in its second dimension.
+    # The number of segments/cosmic rays determine the final products size.
+    # Because each slice is computed indpendently, the number of segments may
+    # differ from segment to segment.  The final output product is created
+    # using the max size for this dimension.  To ensure correct assignment is
+    # done during this step, the second dimension, as well as the row
+    # dimension, must be specified.
+    slope[:, :oslope.shape[1], srow:erow, :] = oslope
+    sigslope[:, :osigslope.shape[1], srow:erow, :] = osigslope
+    var_poisson[:, :ovar_poisson.shape[1], srow:erow, :] = ovar_poisson
+    var_rnoise[:, :ovar_rnoise.shape[1], srow:erow, :] = ovar_rnoise
+    yint[:, :oyint.shape[1], srow:erow, :] = oyint
+    sigyint[:, :osigyint.shape[1], srow:erow, :] = osigyint
+    weights[:, :oweights.shape[1], srow:erow, :] = oweights
+    crmag[:, :ocrmag.shape[1], srow:erow, :] = ocrmag
```

Comment:
I really like the use of mnemonics here.  Much easier to read code.  What's the significance of `shape[1]`?  Is that number of integrations?

Reply:
The significance of `shape[1]` is that it is of variable size.  All but the `crmag` depend on the maximum number of segments in each ramp.  The `crmag` size depends on the maximum number of cosmic rays.  Since it's possible for each slice to have different maximum sizes, the full optional results arrays will be set to the maximum across all slices.  Since it's possible for the different slices to be of different sizes simple assignment can throw an error trying to assign arrays of different sizes.  

For example, it is not possible to do

`crmag[:, :, srow:erow, :] = ocrmag`

if `crmag[:, :, srow:erow, :].shape' is (1, 5, 256, 2048), while `ocrmag.shape` is (1, 2, 256, 2048).  

The shapes must have the same dimensions.  By construction the first, third, and fourth dimensions will be the same.  The second dimension of `crmag` will always be at least as large as the second dimension of `ocrmag`, but it may be larger.  To ensure that this dimension is the same, it needs to be explicitly applied during assignment.